service/service_http_test.go

```diff
@@ -196,6 +196,87 @@ func (s *ServiceHTTPSuite) TestUtf8Message(c *C) {
 		[][]string{[]string(nil), {"Превед Медвед"}, []string(nil), []string(nil)})
 }
 
+// Headers are submitted without a problem
+func (s *ServiceHTTPSuite) TestProduceHeaders(c *C) {
+	if !s.proxyCfg.Kafka.Version.IsAtLeast(sarama.V0_11_0_0) {
+		c.Skip("Headers not supported before Kafka v0.11")
+	}
+
+	svc, err := Spawn(s.cfg)
+	c.Assert(err, IsNil)
+	offsetsBefore := s.kh.GetNewestOffsets("test.1")
+
+	// When
+	req, err := http.NewRequest("POST", "http://_/topics/test.1/messages?key=foo&sync",
+		strings.NewReader("test"))
+	c.Assert(err, IsNil)
+	req.Header.Add("Content-Type", "text/plain")
+	req.Header.Add("X-Kafka-Foo", base64.StdEncoding.EncodeToString([]byte("bar")))
+	rs, err := s.unixClient.Do(req)
+
+	// Then
+	c.Check(err, IsNil)
+	c.Check(rs.StatusCode, Equals, http.StatusOK)
+
+	svc.Stop() // Stop before getting offsets
+	offsetsAfter := s.kh.GetNewestOffsets("test.1")
+	c.Check(offsetsAfter[0], Equals, offsetsBefore[0]+1)
+}
+
+// Invalid base64-encoded headers should produce an error
+func (s *ServiceHTTPSuite) TestProduceInvalidHeaders(c *C) {
+	if !s.proxyCfg.Kafka.Version.IsAtLeast(sarama.V0_11_0_0) {
+		c.Skip("Headers not supported before Kafka v0.11")
+	}
+
+	svc, err := Spawn(s.cfg)
+	c.Assert(err, IsNil)
+	defer svc.Stop()
+
+	// When
+	req, err := http.NewRequest("POST", "http://_/topics/test.1/messages?key=foo&sync",
+		strings.NewReader("test"))
+	c.Assert(err, IsNil)
+	req.Header.Add("Content-Type", "text/plain")
+	req.Header.Add("X-Kafka-Foo", "invalid!")
+	rs, err := s.unixClient.Do(req)
+
+	// Then
+	c.Check(err, IsNil)
+	c.Check(rs.StatusCode, Equals, http.StatusBadRequest)
+
+	body := ParseJSONBody(c, rs).(map[string]interface{})
+	c.Check(body["error"], Equals, "Invalid base64 encoding for header: X-Kafka-Foo")
+}
+
+// Submitting headers when they're not supported by the Kafka version setting
+// should produce an error
+func (s *ServiceHTTPSuite) TestProduceHeadersUnsupported(c *C) {
+	if s.proxyCfg.Kafka.Version.IsAtLeast(sarama.V0_11_0_0) {
```

Comment:
I think it should be 
```
if !s.proxyCfg.Kafka.Version.IsAtLeast(sarama.V0_11_0_0) {
```

Reply:
...fascinating. Why is this passing on newer versions? Or should I just not ask 🙈

Reply:
Oh I guess it skips there which means we're just not getting any signal. Cool - pushing a fix now. Thanks for the pointer!

Reply:
Err wait, no that's actually correct - we _don't_ want to run the test on new versions. I think the actual issue is that gocheck's matcher doesn't behave the way I expected (I think it's implicitly matching on the whole string?)

Reply:
ah, see I should have read the test. LOL